### DISCUSSION TITLE
What is the correct way to add new theme colors?

### DISCUSSION THREAD

**Discussion by icgdoug** [holger1411/understrap#1610] [Q&A]
On BS 4 version of Understrap it used to be simple to add custom colors using the color map and the it was automatically added for use as bg- and text-

Just to confirm this is not changing $primary and $secondary but adding a new color variable such as $tertiary.

In BS 5.1 this no longer works and I haven't been able to find a definitive answer on how to add new colors and have them available using the new utility functions in Understrap Child theme for extending background and text colors.

BS 5.2 is going to release some fixes surrounding this, but in the meantime there should be a way using the existing functions to do this as it means that custom theme colors cannot be used.

**Comment by Mav333rick** [comment]
I was able to add a new color variable $tertiary.

This is what I did:

1. Created a new SASS file called _customized-bootstrap.scss with the following codes shown below.
2. Made sure it was the very first file to be compiled in the SASS importing sequence:
 
`/***************************************
 | New colors
****************************************/
$color-yellow-light:#eced2e;
$color-dark-blue: #20529b;

/***************************************
| Define extra theme color values 
| to be merged with the default Bootstrap
| $theme-colors array.
****************************************/
$custom-theme-colors: (
    "tertiary": $color-yellow-light,
    "dark-blue": $color-dark-blue
);

/***************************************
| Add newly defined theme color to default 
| bootstrap $theme-colors array.
****************************************/
$theme-colors: map-merge($theme-colors , $custom-theme-colors );`

Hope that helps!

**Comment by icgdoug** [comment]
Thanks for your response.

I've been testing this and have been able to get the new colour variables in the system.

There is a extra step that is needed to get these to be used as bg- and text- classes as they are not automatically created.

After:

$theme-colors: map-merge($theme-colors , $custom-theme-colors );`

I added:

.bg-tertiary{background-color: var(--bs-tertiary) !important};
.text-tertiary{color: var(--bs-tertiary) !important};

To get these to be available in the CSS for the new theme colours.

**Comment by IanDelMar** [comment]
Check this out: https://github.com/twbs/bootstrap/issues/35297

**Reply by icgdoug** [reply]
Yes it looks like 5.2 will be releasing fixes. So I think until then we have to work round this.

**Comment by val089** [comment]
Hi,
I tried to change color variable in _theme_variables.scss but nothing changed in theme, when i refresh wordpress.

In the file _theme_variables.scss you can see:

`
// Use this file to overwrite the basic Bootstrap variables and add your own variables
// To overwrite a Bootstrap variable you don´t have to touch the Bootstrap folder.
// Just copy a variable from src/sass/bootstrap4/_variables.scss, paste it here and edit the value.


$purple: #5533ff; // Thats a sample how you could change a BootStrap variable.
$primary: $purple;  // Once we've changed the value, we'll also make it the primary brand color.
`
When I change $purple to $orange I gave an error in terminal.

  
7 │ $primary: $orange;  // Once we've changed the value, we'll also make it the primary brand color.

Do you know how to add custom theme colors?

**Reply by gabrielcastillo** [reply]
Did you run "npm run dev"?

Also, where is "$orange" defined?

**Reply by icgdoug** [reply]
What is the actual error in the terminal?

Have you populated the $orange variable with a hex reference?

**Reply by val089** [reply]
$orange is variable from bootstrap like $purple. There isn't "npm run dev" here. I use npm run watch-bs. 
For instance I try to overwrite class .bg-primary { background-color: red; } but nothing change. Bg-primary class is always purple. Coud you explain me how overwrite classes in understrap?

![image](https://user-images.githubusercontent.com/30775271/161234641-bd0672b9-2079-421a-ac48-272f5206dd9c.png)


This is my first project with understrap and I have problem how to overwrite classes and how to customize understrap theme.

**Reply by gabrielcastillo** [reply]
This is how I would change the color orange;

`
$purple: #5533ff; // Thats a sample how you could change a BootStrap variable.
$primary: $purple;  // Once we've changed the value, we'll also make it the primary brand color.

$orange: #ffa500;
$primary: $orange;`

I defined the "$orange" variable with orange, or a hex value;

Once I save the file, my "npm run watch" compiles the sass.

I could also run "npm run dev" to compile once. But from what you showed, you it looks like you have not defined the "$orange" variable.

The "_child_theme_variables.scss" file is used to define your own variables to override bootstrap default. This file gets loaded before the bootstrap files do. This how we override bootstrap default values.

So if you try to add "$orange: #ffa500;" above "$primary: $orange;" you will see the changes, after you compile the sass files again.

**Reply by gabrielcastillo** [reply]
@val089 If you look into the "child-theme.scss" file, you will see the order of operation.

**Reply by val089** [reply]
@gabrielcastillo I use understrap-main not child. I added as you showed and nothing changed.

`
// Use this file to overwrite the basic Bootstrap variables and add your own variables
// To overwrite a Bootstrap variable you don´t have to touch the Bootstrap folder.
// Just copy a variable from src/sass/bootstrap4/_variables.scss, paste it here and edit the value.


$orange: #ffa500;

$purple: #5533ff; // Thats a sample how you could change a BootStrap variable.
$primary: $orange;  // Once we've changed the value, we'll also make it the primary brand color.
`
Ok, maybe in another way.

How can I change bg color for navbar from purple to orange for example.

![image](https://user-images.githubusercontent.com/30775271/161242064-c86aecb2-8b1a-4a66-a98c-7ff160896be8.png)

**Reply by gabrielcastillo** [reply]
@val089 Here is a quick demo. Hope this helps.

[Loom Video: Change theme variables](https://www.loom.com/share/ded79b750e5e4609b2bdb5a0c3b22c8d)

**Reply by val089** [reply]
@gabrielcastillo Thank you very much for your movie. In your project everything is working beautiful but it doesn't work for me. I dont't why :(. Maybe I have problem in configuration. I download whole project from https://github.com/understrap/understrap and put it in theme folder.
Next I changed browser-sync.config, I added proxy there to start my project after npm run watch-bs. Everythings it starts. But I'dont understand why I havn't got impact on styles.

![image](https://user-images.githubusercontent.com/30775271/161248183-dbd92d19-8a5e-4afd-9657-7c211844a361.png)

**Reply by gabrielcastillo** [reply]
Try without browser sync and just reload the page manually... Also, try clearing your cache. Out of the box everything should be working. I made no changes to the files. I just downloaded the understrap repo, ran npm install and then npm run watch. After that I open the them_variables file and change the primary color. Then once npm re compiled, I reloaded my browser manually...

So with that.. it could be your config. Again try without browser sync. "npm run watch" and then clear your cache and reload your page. While focused on your browser window, press "shift + F5" and you can reload with a fresh cache.

**Reply by val089** [reply]
Dosen't work for me, :(
![image](https://user-images.githubusercontent.com/30775271/161257433-d5ef5790-735c-4a4a-aa1a-5554067725fa.png)

![image](https://user-images.githubusercontent.com/30775271/161257484-5af2133d-c0c4-43b4-a312-4ee66d32b494.png)
![image](https://user-images.githubusercontent.com/30775271/161257536-e9d12f3f-8c9e-461f-9bcf-fbc419ea4f35.png)

**Reply by val089** [reply]
I used npm run watch

**Reply by val089** [reply]
OK, it is resolved. The problem was boostrap 4. My colleague changed in wp-admin bootstrap4 to b5 and everything works.
